### PR TITLE
chore: Renames SDK versions

### PR DESCRIPTION
**Pull request by oarbusi**
## Proposed changes

Renames SDK to a more explicit name.
Suggested in this [PR](https://github.com/mongodb/mongodbatlas-cloudformation-resources/pull/916)

Link to any related issue(s): 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)
- [ ] This change requires a documentation update
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

## Manual QA performed:

- [ ] cfn invoke for each of CRUDL/cfn test
- [ ] Updated resource in  [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples)
- [ ] Published to AWS private registry
- [ ] Used the template in [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples) to create and update a stack in AWS
- [ ] Deleted stack to ensure resources are deleted
- [ ] Created multiple resources in same stack
- [ ] Validated in Atlas UI
- [ ] Included screenshots

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run `make fmt` and formatted my code
- [ ] For CFN Resources: I have released by changes in the private registry and proved by change
  works in Atlas

### CODE REVIEW

File: cfn-resources/util/util.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ import (
 	"github.com/mongodb/mongodbatlas-cloudformation-resources/profile"
 	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/logger"
 	"github.com/mongodb/mongodbatlas-cloudformation-resources/version"
-	atlasSDK "go.mongodb.org/atlas-sdk/v20231115002/admin"
+	admin20231115002 "go.mongodb.org/atlas-sdk/v20231115002/admin"
 	adminLatest "go.mongodb.org/atlas-sdk/v20231115007/admin"
```
Comment (lantoli) [inline cfn-resources/util/util.go:41]: no alias for latest sdk

**Review by lantoli — APPROVED** [review]
LGTM once all checks pass, also a minor comment

File: cfn-resources/util/util.go (hunk 1 above)
Comment (oarbusi) [reply cfn-resources/util/util.go]: Done!

**Review by lantoli — APPROVED** [review]

File: cfn-resources/project/.rpdk-config [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
     "typeName": "MongoDB::Atlas::Project",
     "language": "go",
-    "runtime": "go1.x",
+    "runtime": "provided.al2",
```
Comment (lantoli) [inline cfn-resources/project/.rpdk-config:4]: if this works, should we change it in all resources?

Comment (oarbusi) [reply cfn-resources/project/.rpdk-config:4]: Yes, if it works I will change it for all resources.

Comment (oarbusi) [reply cfn-resources/project/.rpdk-config:4]: It worked well for project resource. I updated it for all other resources

**Review by lantoli — APPROVED** [review]

File: cfn-resources/project/.rpdk-config (hunk 2 above)
Comment (lantoli) [reply cfn-resources/project/.rpdk-config:4]: thx!

**Comment by oarbusi** [comment]
> you have a contract test for data lake pipeline that is failing, not sure if it's a transient error but please first verify that.

It was a transient error. I run it again and it has been successful. Thanks for the heads up

**Pull request merged**